contrib/scalar/scalar.c

```diff
@@ -1018,7 +1063,7 @@ static int cmd_diagnose(int argc, const char **argv)
 	time_t now = time(NULL);
 	struct tm tm;
 	struct strbuf path = STRBUF_INIT, buf = STRBUF_INIT;
-	struct strbuf packs_buf = STRBUF_INIT;
+	struct strbuf packs_buf = STRBUF_INIT, objs_buf = STRBUF_INIT;
```

Comment:
Sorry for spotting this only now, but couldn't we simply reuse `buf`? After the contents are written to `diagnostics.log`, we can call `strbuf_reset(&buf)` and then render the packs information into it. Then write `packs-local.txt`, `strbuf_reset(&buf)` and render the loose objects info into it (and then write `objects-local.txt`).

Reply:
Done.